bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/gemini/gmos/GmosOiwfsProbeArm.scala

```diff
@@ -0,0 +1,142 @@
+package edu.gemini.spModel.gemini.gmos
+
+import java.awt.Shape
+import java.awt.geom.{Point2D, AffineTransform}
+
+import edu.gemini.pot.ModelConverters._
+import edu.gemini.shared.util.immutable.ImPolygon
+import edu.gemini.skycalc.Offset
+import edu.gemini.spModel.core.Coordinates
+import edu.gemini.spModel.inst.{ArmAdjustment, ProbeArmGeometry}
+import edu.gemini.spModel.obs.context.ObsContext
+import edu.gemini.spModel.telescope.IssPort
+
+object GmosOiwfsProbeArm extends ProbeArmGeometry {
+  import edu.gemini.spModel.inst.FeatureGeometry.transformPoint
+
+  val instance = this
+  override protected val guideProbeInstance = GmosOiwfsGuideProbe.instance
+
+  override def geometry: List[Shape] =
+    List(probeArm, pickoffMirror)
+
+  private lazy val probeArm: Shape = {
+    val hm  = PickoffMirrorSize / 2.0
+    val htw = ProbeArmTaperedWidth / 2.0
+
+    val (x0, y0) = (hm,                         -htw)
+    val (x1, y1) = (x0 + ProbeArmTaperedLength, -hm)
+    val (x2, y2) = (x0 + ProbeArmLength,         y1)
+    val (x3, y3) = (x2,                          hm)
+    val (x4, y4) = (x1,                          y3)
+    val (x5, y5) = (x0,                          htw)
+
+    val points = List((x0,y0), (x1,y1), (x2,y2), (x3,y3), (x4,y4), (x5,y5))
+    ImPolygon(points)
+  }
+
+  private lazy val pickoffMirror: Shape = {
+    val (x0, y0) = (-PickoffMirrorSize / 2.0, -PickoffMirrorSize / 2.0)
+    val (x1, y1) = (x0 + PickoffMirrorSize,   y0)
+    val (x2, y2) = (x1,                       y1 + PickoffMirrorSize)
+    val (x3, y3) = (x0,                       y2)
+
+    val points = List((x0,y0), (x1,y1), (x2,y2), (x3,y3))
+    ImPolygon(points)
+  }
+
+  override def armAdjustment(ctx: ObsContext, guideStarCoords: Coordinates, offset: Offset): Option[ArmAdjustment] = {
+    if (ctx == null || offset == null) None
+    else {
+      import ProbeArmGeometry.CanonicalPoint
+      val flip        = if (ctx.getIssPort == IssPort.SIDE_LOOKING) -1.0 else 1.0
+      val posAngle    = ctx.getPositionAngle.toRadians.getMagnitude
+      val offsetPt    = new Point2D.Double(-offset.p.toArcsecs.getMagnitude, -offset.q.toArcsecs.getMagnitude * flip)
+      val guideStarPt = {
+        val baseCoords = ctx.getBaseCoordinates.toNewModel
+        val o = Coordinates.difference(baseCoords, guideStarCoords).offset
+        val (x,y) = (o.p.toArcsecs, o.q.toArcsecs)
+        new Point2D.Double(-x, -y * flip).toCanonicalForm
+      }
+      val wfsOffset   = {
+        ctx.getInstrument match {
+          case gmosn: InstGmosNorth => Some(gmosn.getFPUnit.getWFSOffset)
+          case gmoss: InstGmosSouth => Some(gmoss.getFPUnit.getWFSOffset)
+          case _                    => None
+        }
+      }
+      wfsOffset.map { w =>
+        val angle = armAngle(w, posAngle, guideStarPt, offsetPt)
+        ArmAdjustment(angle, guideStarPt)
+      }
+    }
+  }
+
```

Comment:
In cases like this you're usually better off lifting the nullable values into `Option` and proceeding from there with your entire computation in `Option`. Something like this, perhaps

```scala
override def armAdjustment(ctx0: ObsContext, guideStarCoords: Coordinates, offset0: Offset): Option[ArmAdjustment] =
  for {
    ctx    <- Option(ctx0)
    offset <- Option(offset0)
    w      <- ctx.getInstrument match {
                case gmosn: InstGmosNorth => Some(gmosn.getFPUnit.getWFSOffset)
                case gmoss: InstGmosSouth => Some(gmoss.getFPUnit.getWFSOffset)
                case _                    => None
              }
  } yield {
    val flip = if (ctx.getIssPort == IssPort.SIDE_LOOKING) -1.0 else 1.0
    val posAngle = ctx.getPositionAngle.toRadians.getMagnitude
    val offsetPt = new Point2D.Double(-offset.p.toArcsecs.getMagnitude, -offset.q.toArcsecs.getMagnitude * flip)
    val guideStarPt = {
      val baseCoords = ctx.getBaseCoordinates.toNewModel
      val o = Coordinates.difference(baseCoords, guideStarCoords).offset
      val (x,y) = (o.p.toArcsecs, o.q.toArcsecs)
      new Point2D.Double(-x, -y * flip).toCanonicalForm
    }
    val angle = armAngle(w, posAngle, guideStarPt, offsetPt)
    ArmAdjustment(angle, guideStarPt)
  }
```